src/segmantic/seg/monai_unet.py

```diff
@@ -548,3 +666,185 @@ def print_table(header, vals, indent="\t"):
+
+        print(generalize_img)
+        print(generalize_label)
+
+        test_layers = [(16,32,64,128,256,516), (16,32,64,128), (64,128,256,512,1024)]
+        test_strides = [(2, 2, 2, 2, 2), (2, 2, 2), (2, 2, 2, 2)]
+        print('Pizza 1')
+        for folder in Path(output_dir).iterdir():
+            print(folder)
+            if folder.is_dir() and folder.name == 'Layers[16,32,64,128,256,516]':
+                print('in folder')
+                for fold in folder.iterdir():
+                    #print(fold.name)
+                    for file in fold.iterdir():
+                        #print(file.name)
+                        if file.match('*.ckpt'):
+                            print('start prediction')
+                            predict(model_file=file,
+                                    output_dir=fold,
+                                    test_images=generalize_img,
+                                    test_labels=generalize_label,
+                                    tissue_dict=tissue_dict,
+                                    layers=test_layers[0],
+                                    strides=test_strides[0],
+                                    save_nifti=save_nifti,
+                                    gpu_ids=gpu_ids)
+            elif folder.is_dir() and folder.name == 'Layers[16,32,64,128]':
+                for fold in folder.iterdir():
+                    for file in fold.iterdir():
+                        if file.match('*.ckpt'):
+                            print('start prediction')
+                            predict(model_file=file,
+                                    output_dir=fold,
+                                    test_images=generalize_img,
+                                    test_labels=generalize_label,
+                                    tissue_dict=tissue_dict,
+                                    layers=test_layers[1],
+                                    strides=test_strides[1],
+                                    save_nifti=save_nifti,
+                                    gpu_ids=gpu_ids)
+            elif folder.is_dir() and folder.name == 'Layers[64,128,256,512,1024]':
+                for fold in folder.iterdir():
+                    for file in fold.iterdir():
+                        if file.match('*.ckpt'):
+                            print('start prediction')
+                            predict(model_file=file,
+                                    output_dir=fold,
+                                    test_images=generalize_img,
+                                    test_labels=generalize_label,
+                                    tissue_dict=tissue_dict,
+                                    layers=test_layers[2],
+                                    strides=test_strides[2],
+                                    save_nifti=save_nifti,
+                                    gpu_ids=gpu_ids)
+            else:
+                continue
+
+    kf = KFold(n_splits=n_splits)
```

Comment:
In the `train` I load get the data via `PairedDataSet`. This hides the implementation (globbing and sorting files). I think it would be better to use the PairedDataSet and maybe even move the splitting (using KFold) to the dataset.py, something like 
```
class PairedDataSet():
    ..
  
      @staticmethod
    def kfold_split(num_splits: int) -> List[PairedDataSet]
      ...
```

Those PairedDataSet instances from the split can be saved as json files using `PairedDataSet.dump_dataset`.
Then you also don't have to copy files below, since the split is only done on the list of files names.